[liujiayue314] iP

liujiayue314 opened the pull request:
## Duke task planner

> *"He who fails to plan is planning to fail"* -- Winston Churchill 
> ([Source](https://www.qlik.com/blog/if-you-fail-to-plan-you-are-planning-to-fail-benjamin-franklin))

DukePro is a task planner that is *free*, downloadable it from [here](https://github.com/liujiayue314/ip/releases/tag/A-Jar)!. 
Programmers can also use it in intelliJ, by running `main` method. :technologist:

It frees your mind of having to remember things you need to do. It's
+ Fast to install
+ Suitable for fast typers
+ ~~Easy~~ **Super Easy** to use

It can
- [X] Manage task
- [X] Manage deadline
- [ ] Set reminders (coming soon!)

Currently, the features are 
1. Add/Delete task
2. Mark/Unmark task
3. Search task by keywords


```Ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

liujiayue314 committed:
added greet, echo, exit

liujiayue314 committed:
added: task class + add and list interface

liujiayue314 committed:
added: mark and unmark interface

liujiayue314 committed:
added: Deadline, Event, Todo class

liujiayue314 committed:
Fixed syntax and indentation, added input and expected output txt

liujiayue314 committed:
Added: handling error features

liujiayue314 committed:
Added: delete interface

liujiayue314 committed:
added: Enum

liujiayue314 committed:
Add a saving feature

liujiayue314 committed:
Add date function

liujiayue314 committed:
Merge branch 'branch-Level-7'

liujiayue314 committed:
Resolve conflict between two merges

liujiayue314 committed:
Change the OOP structure

liujiayue314 committed:
Classify into three package: Main, Task, Command

liujiayue314 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

liujiayue314 committed:
Add gradle

jiexuanc commented on src/main/java/Main/Duke.java:
```diff
@@ -0,0 +1,42 @@
+package Main;
```
I think package name should be all lower case.

jiexuanc commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,34 @@
+
+
+    public abstract String toText();
+    abstract public LocalDate getDate();
```
```suggestion
    public abstract LocalDate getDate();
```

jiexuanc commented on src/main/java/Command/DeadlineCommand.java:
```diff
@@ -0,0 +1,28 @@
+import Task.Task;
+
+public class DeadlineCommand extends Command {
+    String description;
```
Maybe having a private access modifiers for attributes would be better?

jiexuanc reviewed (commented):
Overall, aside from missing JavaDocs and some coding violations, the code can be easily understood, variables and methods are well named

liujiayue314 committed:
Add JUnit tests

cjyothika commented on src/main/java/main/Ui.java:
```diff
@@ -0,0 +1,66 @@
+
+public class Ui {
+
+    private static final String indentation = "     ";
```
Perhaps these constant names can be in all caps?

cjyothika commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: "
+                + formatOfDate.format(startDate) + " to: " + formatOfDate.format(endDate) + ")";
```
Good job! This looks like the right indentation for this scenario. Looks like there are no line length issues with your code and you broke your line before an operator.

cjyothika commented on src/main/java/command/MarkCommand.java:
```diff
@@ -0,0 +1,26 @@
+import task.Task;
+
+public class MarkCommand extends Command {
+    int index;
```
Perhaps this class variable should be declared private?

cjyothika commented on src/main/java/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,27 @@
+
+public class UnmarkCommand extends Command{
+
+    int index;
```
Similar to the comment in the MarkCommand class, do consider making this variable private.

cjyothika reviewed (commented):
Generally, your code is readable and formatted well other than a few minor details. Nice work!

liujiayue314 committed:
Add javadocs

liujiayue314 committed:
Comply with a coding standard

liujiayue314 committed:
Add find method

liujiayue314 committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

liujiayue314 committed:
Merge branch 'branch-A-CodingStandard'

liujiayue314 committed:
Merge branch 'branch-Level-9'

liujiayue314 committed:
Update README.md

Apply GFMD in description

liujiayue314 committed:
Checkstyle

liujiayue314 committed:
Checkstyle

liujiayue314 committed:
Merge commit 'dc64d6cade8c03af6b4b41a704c532c29e634569'

liujiayue314 committed:
Add GUI

liujiayue314 committed:
Merge branch 'branch-Level-10'

liujiayue314 committed:
Update README.md

liujiayue314 committed:
Update build.gradle

liujiayue314 committed:
Add event task: inform users when ending date of an event is earlier than starting date of event

Currently, user can add an event with any ending and starting date.

Users should be informed of their mistake and be prompted to change.

Let's check for wrong dates and raise an exception when the mistake is detected.

liujiayue314 committed:
Improve code quality

liujiayue314 committed:
Merge pull request #2 from liujiayue314/branch-A-Assertions

Add event task: inform users when ending date of an event is earlier …

liujiayue314 committed:
Merge branch 'master' into branch-A-CodeQuality

liujiayue314 committed:
Merge pull request #3 from liujiayue314/branch-A-CodeQuality

Branch a code quality

liujiayue314 committed:
Set up CI

liujiayue314 committed:
Improved coding quality

liujiayue314 committed:
Recurring command: able to make task recurring

Previously, users have to manually add task.

Manually adding task can be tedious and users might be forgetful in
adding the task. Recurring command is more user-friendly because users
can leave the task to the task planner.

Let's,
* update input and output to read recurring command
* update the task planner to mark task as not done when the task
refreshes.

liujiayue314 committed:
Merge branch 'branch-Extension'

liujiayue314 committed:
Updated JavaDoc

liujiayue314 committed:
Merge branch 'branch-A-JavaDoc'

liujiayue314 committed:
Update README.md

liujiayue314 committed:
Update checkstyle

liujiayue314 committed:
update test files

liujiayue314 committed:
modified task

liujiayue314 committed:
Updated test files

liujiayue314 committed:
Added Ui.png

liujiayue314 committed:
Add a User Guide

liujiayue314 committed:
updated test files

liujiayue314 committed:
updated user guide

liujiayue314 committed:
minor changes to format

liujiayue314 committed:
Merge branch 'branch-Level-9'

liujiayue314 committed:
Update README.md
